Python/thread.c

```diff
@@ -114,12 +114,9 @@ PyThread_set_stacksize(size_t size)
 #endif
 }
 
-#ifndef Py_HAVE_NATIVE_TLS
```

Comment:
I suggest  to keep the #ifndef and emit a compiler error (#error "...") here with an helpful error message, like the http://bugs.python.org/issue30832 URL.

Or maybe, chain the following two if and add the error in a new #else block?

```
#ifdef _POSIX_THREADS
#define PYTHREAD_NAME "pthread"
#include "thread_pthread.h"
#endif

#ifdef NT_THREADS
#define PYTHREAD_NAME "nt"
#include "thread_nt.h"
#endif
```